records_mover/records/schema/field/__init__.py

```diff
@@ -64,9 +64,9 @@ def __init__(self,
     def refine_from_series(self,
                            series: 'Series',
                            total_rows: int,
-                           rows_sampled: int) -> None:
+                           rows_sampled: int) -> 'RecordsSchemaField':
```

Comment:
This previously mutated the series passed in; it now either returns the original or a new object.  Other methods in this chain of calls are changed similarly.

Reply:
I love immutability.